ballerina/client_endpoint.bal

```diff
@@ -154,6 +176,38 @@ isolated function generateMethodId(string? pkgName, string svcName, string rpcNa
     return methodID;
 }
 
+// Initialize the client auth handler based on the provided configurations
+isolated function initClientAuthHandler(ClientAuthConfig authConfig) returns ClientAuthHandler {
+    if authConfig is CredentialsConfig {
+        ClientBasicAuthHandler handler = new(authConfig);
+        return handler;
+    } else if authConfig is BearerTokenConfig {
+        ClientBearerTokenAuthHandler handler = new(authConfig);
+        return handler;
+    } else if authConfig is JwtIssuerConfig {
+        ClientSelfSignedJwtAuthHandler handler = new(authConfig);
+        return handler;
+    } else {
+        // Here, `authConfig` is `OAuth2GrantConfig`
+        ClientOAuth2Handler handler = new(authConfig);
+        return handler;
+    }
+}
+
+// Enriches the request using the relevant client auth handler
+isolated function enrichHeaders(ClientAuthHandler clientAuthHandler, map<string|string[]> headers) returns map<string|string[]>|ClientAuthError {
+    if (clientAuthHandler is ClientBasicAuthHandler) {
```

Comment:
Shall we remove these unnecessary parentheses?

Reply:
Removed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/350/commits/bba296b92881ca062ad2b802684cc6f690bcba4f